podpac/datalib/soilscape.py

```diff
@@ -0,0 +1,663 @@
+
+    Arguments
+    ---------
+    site : str
+        SoilSCAPE site, e.g. 'Canton_OK'
+    time : array, datetime64, str
+        datetime(s). Default is the current time.
+    depth : float, array
+        depth(s). Default: [4, 13, 30] (all available depths)
+
+    Returns
+    -------
+    coords : Coordinates
+        Coordinates with (lat_lon) for all nodes at the site and the given time and depth
+    """
+
+    if site not in NODES:
+        raise ValueError("site '%s' not found" % site)
+
+    if time is None:
+        time = np.datetime64(datetime.datetime.now())  # now
+
+    if depth is None:
+        depth = [4, 13, 30]  # all
+
+    lats = []
+    lons = []
+    for node in NODES[site]:
+        try:
+            lat, lon = get_node_location(node)
+        except:
+            _logger.exception("Could not get coordinates for '%s' node '%s'" % (NODE2SITE[node], node))
+            continue
+        lats.append(lat)
+        lons.append(lon)
+
+    return podpac.Coordinates([[lats, lons], time, depth], dims=["lat_lon", "time", "alt"], crs=CRS)
+
+
+class SoilSCAPEFile(podpac.data.Dataset):
+    """ SoilSCAPE datasource from file. """
+
+    data_key = ["soil_moisture", "moisture_flag"]
+    alt_key = "depth"
+
+    @property
+    def lat(self):
+        return self.dataset.lat.item()
+
+    @property
+    def lon(self):
+        return self.dataset.lon.item()
+
+    @property
+    def physicalid(self):
+        # note: this should be the same as the node number
+        return self.dataset.physicalid.item()
+
+    def get_coordinates(self):
+        coordinates = super(SoilSCAPEFile, self).get_coordinates()
+        coordinates.set_trait("crs", CRS)
+        return coordinates
+
+
+class SoilSCAPENode(SoilSCAPEFile):
+    """ SoilSCAPE 20min soil moisture for a particular node.
+
+        Data is loaded from the THREDDS https fileserver.
+
+        Attributes
+        ----------
+        site : str
+            SoilSCAPE site, e.g. 'Canton_OK'.
+        node : int
+            SoilSCAPE node id.
+    """
+
+    site = tl.Enum(list(NODES)).tag(attr=True)
+    node = tl.Int().tag(attr=True)
+    cache_dataset = tl.Bool(True)
+
+    _repr_keys = ["site", "node"]
+
+    @tl.default("dataset_expires")
+    def default_dataset_expires(self):
+        if self.site in EXPIRED:
+            return None
+        else:
+            return "1,D"
+
+    @tl.validate("node")
+    def _validate_node(self, d):
+        if d["value"] not in NODES[self.site]:
+            raise ValueError("Site '%s' does not have a node n%d" % (self.site, d["value"]))
+
+        return d["value"]
+
+    @property
+    def source(self):
+        return "{base_url}/{filename}.nc".format(base_url=SOILSCAPE_FILESERVER_BASE, filename=self.filename)
+
+    @property
+    def filename(self):
+        return "soil_moist_20min_{site}_n{node}".format(site=self.site, node=self.node)
+
+
+class SoilSCAPE20min(podpac.core.compositor.compositor.BaseCompositor):
```

Comment:
@mpu-creare I used BaseCompositor here because the composite method needs to add dimensions to the source data. I'm not sure if this can/should be factored out.